docs/rules/prefer-destructuring.md

```diff
@@ -0,0 +1,82 @@
+# Prefer destructuring from arrays and objects (prefer-destructuring)
+
+With JavaScript ES6, a new syntax was added for creating variables from an array index of object property, called [destructuring](#further-reading).  This rule enforces usage of destructuring instead of accessing a property through a member expression.
+
+## Rule Details
```

Comment:
Can you add a paragraph that array destructuring is only enforced if array index is a literal? The last "correct" example is slightly confusing without this note. As in, this rule will flag this code: `var a = array[100];` but will not flag this code: `var a = array[myVar];`